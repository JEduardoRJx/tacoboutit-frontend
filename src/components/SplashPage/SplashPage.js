import React, { Component } from 'react';
import { Text, View, ActivityIndicator, Dimensions, StyleSheet } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import * as Font from 'expo-font';


export default class SplashPage extends Component {
  state = {
    search: '',
    isLoading: true,
  };

  componentDidMount = async () => {
    await Font.loadAsync({
      'molle': require('../../../assets/fonts/Molle-Regular.ttf'),
    })
    this.setState({ isLoading: false });
  }

  render() {
    const fontFamily = this.state.isLoading ? null : { fontFamily: 'molle' };
    const dynamicStyles = [styles.header, fontFamily];
    return (
      <LinearGradient 
        colors={["#F0CB35", "#D56C2C", "#C02425"]}  
        style={styles.splashContainer}
      >
        <View style={styles.headerContainer}>
          <Text style={dynamicStyles}>Taco</Text> 
          <Text style={dynamicStyles}>BoutIt</Text> 
        </View>
        <ActivityIndicator size="large" color="#fff" animating={this.props.isLoading}/>
      </LinearGradient>
    );
  }
}
const styles = StyleSheet.create({
  splashContainer: {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
    position: 'absolute',
    zIndex: 5,
  },
  header: {
    fontSize: 100,
    color: 'white',
  },
  headerContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  }
})

