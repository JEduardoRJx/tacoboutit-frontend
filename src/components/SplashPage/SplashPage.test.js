import React from 'react';
import renderer from 'react-test-renderer';

import SplashPage from '../SplashPage/SplashPage';

describe('SplashPage', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<SplashPage />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});