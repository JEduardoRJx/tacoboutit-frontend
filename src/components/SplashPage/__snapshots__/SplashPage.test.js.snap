// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashPage renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4293970741,
        4292176940,
        4290782245,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 1334,
      "justifyContent": "space-around",
      "position": "absolute",
      "width": 750,
      "zIndex": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 100,
          },
          null,
        ]
      }
    >
      Taco
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 100,
          },
          null,
        ]
      }
    >
      BoutIt
    </Text>
  </View>
  <ActivityIndicator
    animating={true}
    color="#fff"
    hidesWhenStopped={true}
    size="large"
  />
</ViewManagerAdapter_ExpoLinearGradient>
`;
